### PR TITLE
fixed if statement flow for env and volume mounts

### PR DESCRIPTION
**Pull request by LockedThread**
Yeah, so... in my [last PR](https://github.com/apollographql/router/pull/1245) I accidentally broke some if logic for environmental variables and volume mounts with the pre-existing code. This code will fix it. Lmk what you need and I'll get it done.

### CODE REVIEW

**Comment by LockedThread** [comment]
I have tested the forked chart and it is working for me in production right now. If you find any issues or would like any changes lmk.

File: helm/chart/router/templates/deployment.yaml [hunk 1]
```diff
@@ -99,10 +103,12 @@ spec:
             {{- toYaml .Values.resources | nindent 12 }}
           {{- if or .Values.router.configuration  .Values.extraVolumeMounts }}
           volumeMounts:
+            {{- if or .Values.router.configuration }}
             - name: router-configuration
```
Comment (garypen) [inline helm/chart/router/templates/deployment.yaml:107]: I don't think the `or` is required.

**Review by garypen — CHANGES_REQUESTED** [review]
Looks good. One small change please.

File: helm/chart/router/templates/deployment.yaml (hunk 1 above)
Comment (LockedThread) [reply helm/chart/router/templates/deployment.yaml:107]: yeah woops I just copy the previous line and that happened, still worked in testing tho... odd

**Comment by LockedThread** [comment]
resolved

**Comment by LockedThread** [comment]
@garypen I know this is off topic but could you look into https://github.com/apollographql/router/issues/75 and the lack of documentation with this being supposedly implemented? Outbound header pass through is super important when dealing with cookies.

**Review by garypen — APPROVED** [review]

**Comment by garypen** [comment]
> @garypen I know this is off topic but could you look into #75 and the lack of documentation with this being supposedly implemented? Outbound header pass through is super important when dealing with cookies.

I'll bring it up in tomorrow's team meeting.

**Comment by LockedThread** [comment]
> 
I found a relevant commit you might be able to bring up:
https://github.com/apollographql/router/commit/b1714a9ef56154f53704b9dbd9a853a6b3836f92#diff-c8b5b6a5805410d13413a02adddd806bc32dacdaedf1e80d14e68c95dd6bb6feR232

The documentation has been changed quite a bit since then and it seems like the features mentioned have been erased... odd.

If you could please let me know what happens that would be great. 
contact@warrensnipes.dev

**Comment by LockedThread** [comment]
> > @garypen I know this is off topic but could you look into #75 and the lack of documentation with this being supposedly implemented? Outbound header pass through is super important when dealing with cookies.
> 
> I'll bring it up in tomorrow's team meeting.

I found a relevant commit you might be able to bring up:
https://github.com/apollographql/router/commit/b1714a9ef56154f53704b9dbd9a853a6b3836f92#diff-c8b5b6a5805410d13413a02adddd806bc32dacdaedf1e80d14e68c95dd6bb6feR232

The documentation has been changed quite a bit since then and it seems like the features mentioned have been erased... odd.

If you could please let me know what happens that would be great. 
contact@warrensnipes.dev

**Comment by LockedThread** [comment]
Also check this out https://community.apollographql.com/t/apollo-router-forward-subgraph-headers-to-clients/3649

**Comment by garypen** [comment]
@LockedThread  I raised #1284. You can follow that issue to see how it progresses.

**Pull request merged**